invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/ShareModal.js

```diff
@@ -6,59 +6,255 @@
 // under the terms of the MIT License; see LICENSE file for more details.
 
 import React, { Component } from "react";
-import { Icon, Modal, Tab, Container } from "semantic-ui-react";
+import { Icon, Modal, Tab, Container, Button } from "semantic-ui-react";
 import { i18next } from "@translations/invenio_app_rdm/i18next";
 import PropTypes from "prop-types";
 import { LinksTab } from "./AccessLinks/LinksTab";
 import { AccessRequestsTab } from "./AccessRequests/AccessRequestsTab";
-import { AccessUsers } from "./AccessUsers/AccessUsers";
+import { AccessUsersGroups } from "./AccessUsersGroups/AccessUsersGroups";
+import { GroupsApi } from "@js/invenio_communities/api";
+// import { UsersApi } from "@js/invenio_communities/api"; uncomment together with the code below
+import { withCancel } from "react-invenio-forms";
+import { http } from "react-invenio-forms";
 
 export class ShareModal extends Component {
   constructor(props) {
     super(props);
     const { record } = this.props;
     this.state = {
       record: record,
+      usersResults: undefined,
+      groupsResults: undefined,
+      linksResults: undefined,
+      usersLoading: false,
+      groupsLoading: false,
+      linksLoading: false,
+      usersError: undefined,
+      groupsError: undefined,
+      linksError: undefined,
+      activeTab: 0,
     };
   }
 
+  handleTabChange = (e, data) => {
+    this.setState({ activeTab: data.activeIndex });
+  };
+
   handleRecordUpdate = (updatedRecord) => {
+    const { record } = this.state;
+    updatedRecord.expanded = record.expanded;
     this.setState({ record: updatedRecord });
   };
 
+  userType = "user";
+  groupType = "group";
+  linksType = "links";
+
+  updatedCallback = (id, permission, entityType) => {
+    let results;
+    if (entityType === this.userType) {
+      const { usersResults } = this.state;
+      results = usersResults;
+    }
+    if (entityType === this.groupType) {
+      const { groupsResults } = this.state;
+      results = groupsResults;
+    }
+    if (entityType === this.linksType) {
+      const { linksResults } = this.state;
+      results = linksResults;
+    }
+
+    results.forEach((result) => {
+      if (result.id === id) {
+        result.permission = permission;
+      }
+    });
+  };
+
+  addedOrDeletedCallback = async (callbackEndpoint, entityType) => {
+    if (entityType === this.userType || entityType === this.groupType)
+      await this.fetchUsersGroups(callbackEndpoint, entityType, true);
+    if (entityType === this.linksType) await this.fetchLinks(callbackEndpoint, true);
+  };
```

Comment:
for readability of the code we could use `isUserType` and `isLinksType` pattern (precalculated variables) :
```javascript
const isUserGroupType = entityType === this.userType || entityType === this.groupType

const isLinksType = (entityType === this.linksType)

if (isUserGroupType){
...
}
```